miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditServiceRecordController.java

```diff
@@ -0,0 +1,147 @@
+  
+  @Autowired
+  private RequestManager requestManager;
+
+  @Autowired
+  private FilesManager filesManager;
+
+  @Autowired
+  private DataObjectFactory dataObjectFactory;
+
+  public void setRequestManager(RequestManager requestManager) {
+    this.requestManager = requestManager;
+  }
+
+  public void setFilesManager(FilesManager filesManager) {
+    this.filesManager = filesManager;
+  }
+
+  public void setDataObjectFactory(DataObjectFactory dataObjectFactory) {
+    this.dataObjectFactory = dataObjectFactory;
+  }
+  
+  public Map<Integer, String> populateServiceRecordFiles(SequencerServiceRecord record) throws IOException {
+    if (record.getId() != AbstractSequencerServiceRecord.UNSAVED_ID) {
+      Map<Integer, String> fileMap = new HashMap<Integer, String>();
+      for (String s : filesManager.getFileNames(SequencerServiceRecord.class, String.valueOf(record.getId()))) {
+        fileMap.put(s.hashCode(), s);
+      }
+      return fileMap;
+    }
+    return Collections.emptyMap();
+  }
+  
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> maxLengths() throws IOException {
+    return requestManager.getServiceRecordColumnSizes();
+  }
+
+  @RequestMapping(value = "/{recordId}", method = RequestMethod.GET)
+  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, ModelMap model) throws IOException {
+    SequencerServiceRecord sr = requestManager.getSequencerServiceRecordById(recordId);
+    if (sr != null) {
+      model.put("serviceRecord", sr);
+      model.put("serviceRecordFiles", populateServiceRecordFiles(sr));
+    } else {
+      throw new IOException("Cannot retrieve the requested Service Record");
+    }
+    return new ModelAndView("/pages/editServiceRecord.jsp", model);
+  }
+  
+  @RequestMapping(value = "/new/{sequencerReferenceId}", method = RequestMethod.GET)
+  public ModelAndView newServiceRecord(@PathVariable(value = "sequencerReferenceId") Long sequencerReferenceId, ModelMap model) throws IOException {
+    SequencerReference sequencer = requestManager.getSequencerReferenceById(sequencerReferenceId);
+    if (sequencer == null) {
+      throw new IOException("Cannot retrieve the requested Sequencer Reference");
+    }
+    SequencerServiceRecord record = dataObjectFactory.getSequencerServiceRecord();
+    record.setSequencerReference(sequencer);
+    model.put("serviceRecord", record);
+    return new ModelAndView("/pages/editServiceRecord.jsp", model);
+  }
+  
+  @RequestMapping(method = RequestMethod.POST)
+  public String processSubmit(@ModelAttribute("serviceRecord") SequencerServiceRecord record, ModelMap model, SessionStatus session)
+      throws IOException {
+    try {
+      requestManager.saveSequencerServiceRecord(record);
+      session.setComplete();
+      model.clear();
+      return "redirect:/miso/stats/sequencer/servicerecord/" + record.getId();
+    } catch (IOException ex) {
```

Comment:
I was following the pattern of other EditXControllers, but as you say, the catch/log/throw is a bit pointless, since it will already be logged. I'll remove the try/catch block for now.

MISO throws IOExceptions for a lot of different reasons. In the future, it'd be nice to throw more varied exception types, preferably annotated with Spring's ResponseStatus, so more friendly and helpful error pages could be easily displayed.